cloud/openstack/os_compute_quotas.py

```diff
@@ -0,0 +1,134 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+
+
+DOCUMENTATION = '''
+---
+module: os_compute_quotas
+short_description: Update/Delete compute quotas in a OpenStack tenant.
+extends_documentation_fragment: openstack
+author: "Ghe Rivero (@GheRivero) and Andrea Mercanti (@mercantiandrea)"
+version_added: "2.3"
+description:
+   - Update or delete compute quotas in a OpenStack tenant.
+options:
+  project:
+    description:
+      - The name or id of the project to which the quota should be assigned
+    required: true
+  state:
+    description:
+      - Should the resource be present or absent
+    choices: ['present', 'absent']
+    required: false
+    default: present
+  quota:
+    description:
+      - Dictionary containing the quotas and values you want to update
+    required: false
+
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Update the quota of the tenant 'demo'
+- os_compute_quotas:
+    cloud: mycloud
+    project: demo
+    state: present
+    quota:
+      instances: 33
+      cores: 33
+      ram: 33333
+      floating_ips: 33
+      fixed_ips: 33
+      metadata_items: 33
+      injected_files: 33
+      injected_file_content_bytes: 33333
+      injected_file_path_bytes: 333
+      key_pairs: 33
+      security_groups: 33
+      security_group_rules: 33
+      server_groups: 33
+      server_group_members: 33
+
+# Delete quotas for the tenant 'demo' and back to defaults
+- os_compute_quotas:
+    cloud: mycloud
+    project: demo
+    state: absent
+'''
+
+RETURN = '''
+quotas:
+    description: Dictionary listing all the compute quotas and their values for the project
+    returned: On success
+    type: dictionary
+
+'''
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        project = dict(required=False),
+        quota = dict(default=None, type='dict'),
+        state = dict(default='present', choices=['absent', 'present']),
+    )
+
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec, **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    if (StrictVersion(shade.__version__) < StrictVersion('1.8.0')):
+        module.fail_json(msg="To manage compute quotas, the installed version of"
+                             "the shade library MUST be >=1.8.0")
+
+    try:
+        cloud = shade.operator_cloud(**module.params)
+
+        old_quotas = cloud.get_compute_quotas(module.params['project'])
+
+        if module.params['state'] == 'present':
```

Comment:
There is no support for `check` mode here, nor is there a method to prevent setting quotas just to see if anything has changed.

A method to examine the user specified quotas against what's reported in old_quotas should be added, as that can determine if a change is required, thus supporting check mode, and prevent unnecessary API calls.